db/migrate/20170821035759_add_original_to_project.rb

```diff
@@ -0,0 +1,8 @@
+class AddOriginalToProject < ActiveRecord::Migration
+  def change
+    add_column :projects, :original_id, :integer, polymorphic: true, foreign_key: { references: :projects}
+    # add_reference :projects, :original
+    # add_foreign_key :projects, :original, column: :original_id
+    # add_index :projects, [:id, :original_id]
```

Comment:
Why are these commented?

Here would be my thoughts:
* You are right to have a foreign key on original_id. 
* original_id should also have an index.
* What is the compound index for?
* What does `polymorphic: true` do for `add_column`? I've only ever seen it with `add_reference`? I don't think we need an `original_type` column, do we?